plugin-rapl/README.md

```diff
@@ -1,6 +1,52 @@
-# RAPL plugin
+# Plugin RAPL
 
-Collects the energy consumption of the CPU and other RAPL domains.
+The RAPL plugin creates an Alumet **source** that collects measurements of processor energy usage via [RAPL interfaces](https://www.intel.com/content/www/us/en/developer/articles/technical/software-security-guidance/advisory-guidance/running-average-power-limit-energy-reporting.html), such as perf-events and powercap.
 
-Currently, this plugin only works on Linux, because it relies on some abstractions provided by the Linux kernel over RAPL.
-Using MSR registers directly is tricky, hard to maintain, and does not offer any performance benefit.
+## ⚙️ Requirements
+
+- RAPL-compatible processor
+- Linux (the plugin relies on abstractions provided by the kernel - perf-events and powercap)
+- [Set perf_event_paranoid to 0](https://github.com/alumet-dev/packaging/blob/main/docker/README.md#perfmon-usage) (Only for perf-events usage).
+- [Capabilities](https://github.com/alumet-dev/packaging/blob/main/docker/README.md) (Only for containers).
+
+## 📊Metrics
+
+Here are the metrics collected by the plugin source.
+
+|Name|Type|Unit|Description|Attributes|More information|
+|----|----|----|-----------|----------|-----------------|
+|`rapl_consumed_energy`|Counter Diff|joule|Energy consumed since the previous measurement|[domain](#domain-attribute-rapl_consumed_energy)||
+
+### Domain attribute (rapl_consumed_energy)
+
+The `rapl_consumed_energy` metric is classified by the `domain` attribute. The possible domain values are:
+
+|Domain|Description|
+|------|-----------|
+|`platform`|the entire server|
+|`package`|the CPU cores, the iGPU, the L3 cache and the controllers|
+|`pp0`|the CPU cores|
+|`pp1`|the iGPU|
+|`dram`|the RAM attached to the processor|
+
+## 🛠️Configuration
+
+Here is a configuration example of the RAPL plugin. It's part of the Alumet configuration file (eg: `alumet-config.toml`).
+
+```toml
+[plugins.rapl]
+# Interval between two RAPL measurements.
+poll_interval = "1s"
+# Interval between two flushing of RAPL measurements.
+flush_interval = "5s"
+# Set to true to disable perf-events and always use the powercap sysfs.
+no_perf_events = false
+```
+
+## Should I use perf-events or powercap ?
+
+Both interfaces provide similar energy consumption data, but we recommend using perf-events for lower measurement overhead (especially in high-frequency polling scenarios).
+
+Note: perf-events requires additional Linux capabilities, while powercap doesn't.
```

Comment:
True, but powercap require read permissions for `/sys/class/powercap/intel-rapl*` and all subfolders.

Reply:
You can find more insight about the required permissions and capabilities in the source code of the plugin itself.